playbooks/roles/deploy-osh/tasks/main.yml

```diff
@@ -151,22 +151,19 @@
   tags:
     - run
 
-- name: Create .config/openstack directory
+- name: Ensure /etc/openstack directory exists
+  become: yes
   file:
-    path: /root/.config/openstack
+    path: "{{ item }}"
     state: directory
-    owner: root
-    group: root
-    mode: '0755'
-  tags:
-    - run
+  with_items:
+    - /etc/openstack
 
 - name: Save default clouds.yaml file
+  become: yes
   template:
-    src: clouds.yaml.j2
-    dest: /root/.config/openstack/clouds.yaml
-    owner: root
-    mode: '0640'
```

Comment:
why did you remove the root/mode setting? I think it is better to be explicit?

Reply:
I think this is for the user that runs the playbook to have access to the clouds.yaml that is created? That user can be root, or "stack" for example. maybe eventually (in product version) we need a group that can access the conf files that have secrets in it.

Reply:
Yeah, it's just copied from the airship part

Reply:
as you're running this as root (see become: yes), we should be careful with the mode and owner settings. please just add the owner: and mode: lines that you added back (I would assume it actually needs to be 0644 for non-root to be able to read the clouds.yaml). 

please note that you can split secrets into a separate secret yaml, that could be stored under the ~root user as before so that generic settings other than the password are available to everyone.